Automattic/wp-calypso
# Store: Move fetching general settings to the data layer

This PR updates general settings loading to use the data layer so that we can benefit from things like [retrying network requests on failure](https://github.com/Automattic/wp-calypso/tree/6733ba43d1d91f8f45edaf8313107036bc3871aa/client/state/data-layer/wpcom-http/pipeline/retry-on-failure). Thanks to Kevin for pairing with me on this today. I had to make a few updates after we split, so I've added you as a reviewer.

To Test:
* Run `npm run test-client client/extensions/woocommerce/state` and make sure all tests pass.
* Go to `http://calypso.localhost:3000/store/:site` and click around to various parts of the UI that use 'general settings'. This includes the currency symbol on the product form, our settings pages (taxes, payments, shipping), the address component on the settings pages, and the various dashboard views.
* Make sure that data loads in correctly on these pages.
* You can also monitor the Redux console for `WOOCOMMERCE_SETTINGS_GENERAL_REQUEST_SUCCESS` and see that the correct data was loaded in.
